[libsquish] build failure on <any>-android with ndk 27

### Operating system

Linux

### Compiler

Clang

### Steps to reproduce the behavior

```Shell
export ANDROID_NDK_HOME=$HOME/Downloads/android-ndk-r27/
run `vcpkg install libsquish:arm-android`
```


### Failure logs
```
CMake Warning (dev) at CMakeLists.txt:12 (PROJECT):
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at android-ndk-r27/build/cmake/flags.cmake:46 (if):
  Policy CMP0057 is not set: Support new IN_LIST if() operator.  Run "cmake
  --help-policy CMP0057" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  IN_LIST will be interpreted as an operator when the policy is set to NEW.
  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  /usr/share/cmake/Modules/Platform/Android-Clang.cmake:23 (include)
  /usr/share/cmake/Modules/Platform/Android-Clang-C.cmake:1 (include)
  /usr/share/cmake/Modules/CMakeCInformation.cmake:48 (include)
  CMakeLists.txt:12 (PROJECT)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at Downloads/android-ndk-r27/build/cmake/flags.cmake:46 (if):
  if given arguments:

    "hwaddress" "IN_LIST" "ANDROID_SANITIZE"

  Unknown arguments specified
Call Stack (most recent call first):
  /usr/share/cmake/Modules/Platform/Android-Clang.cmake:23 (include)
  /usr/share/cmake/Modules/Platform/Android-Clang-C.cmake:1 (include)
  /usr/share/cmake/Modules/CMakeCInformation.cmake:48 (include)
  CMakeLists.txt:12 (PROJECT)
```

### Additional context

https://github.com/android/ndk/issues/2032

--- a/ports/libsquish/cmake-version.patch
+++ b/ports/libsquish/cmake-version.patch
@@ -1,0 +1,16 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index a36e574..6f137d5 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -9,9 +9,9 @@
+ #   Unix and VS: SSE2 support is enabled by default
+ #    use BUILD_SQUISH_WITH_SSE2 and BUILD_SQUISH_WITH_ALTIVEC to override
+
+-PROJECT(squish)
++CMAKE_MINIMUM_REQUIRED(VERSION 3.5)
+
+-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.3)
++PROJECT(squish)
+
+ OPTION(BUILD_SQUISH_WITH_OPENMP "Build with OpenMP." ON)
+ 

--- a/ports/libsquish/portfile.cmake
+++ b/ports/libsquish/portfile.cmake
@@ -7,6 +7,7 @@ vcpkg_from_sourceforge(
     PATCHES
         fix-export-symbols.patch
         export-target.patch
+        cmake-version.patch
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS

--- a/ports/libsquish/vcpkg.json
+++ b/ports/libsquish/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libsquish",
   "version": "1.15",
-  "port-version": 13,
+  "port-version": 14,
   "description": "Open source DXT compression library.",
   "homepage": "https://sourceforge.net/projects/libsquish",
   "license": "MIT",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5062,7 +5062,7 @@
     },
     "libsquish": {
       "baseline": "1.15",
-      "port-version": 13
+      "port-version": 14
     },
     "libsrt": {
       "baseline": "1.5.3",

--- a/versions/l-/libsquish.json
+++ b/versions/l-/libsquish.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "eb13fe850ce51bd142d19b0662ba0d990efb5d77",
+      "version": "1.15",
+      "port-version": 14
+    },
+    {
       "git-tree": "b6a8a3e697c1cfcd378bfe725d5994ac9e9dc6d3",
       "version": "1.15",
       "port-version": 13

